sympy/sets/sets.py

```diff
@@ -1281,6 +1357,16 @@ def _measure(self):
             parity *= -1
         return measure
 
+    @property
+    def _kind(self):
+        args = self._helper_kind(self.args, S.EmptySet)
+        if not args:
+            return SetKind()
+        elif all(i == args[0] for i in args):
+            return args[0]
+        else:
+            return SetKind(UndefinedKind)
```

Comment:
It would be better if the `_helper_kind` method did all of this. It could be called `_union_setkinds` and then this method could just do:
```
return self._union_setkinds(self.args))
```
Then you wouldn't need to duplicate this code in other places.